shell/CODE_STYLE.md

```diff
@@ -0,0 +1,251 @@
+# Shell Script Code Style Guide
+
+This document outlines the coding standards and best practices for shell scripts in the Dataverse repository.
```

Comment:
Just a high level comment...

@poikilotherm added shellcheck and shellspec to the Dataverse repo: https://github.com/IQSS/dataverse/commit/f8bf73479708a0d1cfb6882db9a118e12d70d34d

I wouldn't be opposed to adding either or both here if you want.